pinot-core/src/main/java/com/linkedin/pinot/core/indexsegment/generator/SegmentGeneratorConfig.java

```diff
@@ -144,6 +147,44 @@ public SegmentGeneratorConfig(SegmentGeneratorConfig config) {
     _onHeap = config._onHeap;
   }
 
+  public SegmentGeneratorConfig(TableConfig tableConfig, Schema schema) {
+    Preconditions.checkNotNull(schema);
+    _schema = schema;
+
+    if (tableConfig != null) {
```

Comment:
I think it's better to put null check like `if (tableConfig == null) return;` in order to avoid the nested if statements.